### PR TITLE
refactor(ext/node): add more methods to 'NodeFs' trait

### PR DESCRIPTION
**Pull request by bartlomieju**
Added more methods to `ext/node/clippy.toml` that are not allowed
to be used in the crate.

Prerequisite for https://github.com/denoland/deno/pull/18544

### CODE REVIEW

File: ext/node/clippy.toml [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
   { path = "std::path::Path::exists", reason = "File system operations should be done using NodeFs trait" },
+  { path = "std::path::Path::canonicalize", reason = "File system operations should be done using NodeFs trait" },
+  { path = "std::path::Path::is_dir", reason = "File system operations should be done using NodeFs trait" },
+  { path = "std::path::Path::is_file", reason = "File system operations should be done using NodeFs trait" },
+  { path = "std::path::Path::is_symlink", reason = "File system operations should be done using NodeFs trait" },
+  { path = "std::path::Path::metadata", reason = "File system operations should be done using NodeFs trait" },
+  { path = "std::path::Path::read_dir", reason = "File system operations should be done using NodeFs trait" },
+  { path = "std::path::Path::read_link", reason = "File system operations should be done using NodeFs trait" },
+  { path = "std::path::Path::symlink_metadata", reason = "File system operations should be done using NodeFs trait" },
+  { path = "std::path::Path::try_exists", reason = "File system operations should be done using NodeFs trait" },
```
Comment (dsherret) [inline ext/node/clippy.toml:12]: Do we need all of these for `PathBuf` too?

**Review by dsherret — APPROVED** [review]
LGTM

**Review by lucacasonato — COMMENTED** [review]
You shouldn't return `std::fs::Metadata` from the trait methods. It is not implementable in a VFS, because you can't construct a `std::fs::Metadata`.

**Comment by bartlomieju** [comment]
Good point. I'll change it

File: ext/node/clippy.toml (hunk 1 above)
Comment (bartlomieju) [reply ext/node/clippy.toml:12]: Added

**Pull request merged**